### PR TITLE
Fix scaladoc warnings

### PR DESCRIPTION
**Pull request by rozza**
JAVA-4029

### CODE REVIEW

File: driver-scala/src/main/scala/org/mongodb/scala/MongoCollection.scala [hunk 1]
```diff
@@ -151,19 +151,19 @@ case class MongoCollection[TResult](private val wrapped: JMongoCollection[TResul
    * Counts the number of documents in the collection.
    *
    * '''Note:'''
-   * For a fast count of the total documents in a collection see [[estimatedDocumentCount()]]
+   * For a fast count of the total documents in a collection see [[estimatedDocumentCount()*]]
```
Comment (jyemin) [inline driver-scala/src/main/scala/org/mongodb/scala/MongoCollection.scala:154]: What does the `*` signify?

File: driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala [hunk 2]
```diff
@@ -174,9 +174,9 @@ trait ObservableImplicits {
     val observable = obs
 
     /**
-     * Collects the [[Observable]] results and converts to a [[scala.concurrent.Future]].
+     * Collects the [[Observable]] results and converts to a scala.concurrent.Future`.
```
Comment (jyemin) [inline driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala:177]: Missing starting backtick?

File: driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala [hunk 3]
```diff
@@ -174,9 +174,9 @@ trait ObservableImplicits {
...
      *
-     * Automatically subscribes to the `Observable` and uses the [[collect]] method to aggregate the results.
+     * Automatically subscribes to the `Observable` and uses the [[Observable.collect]] method to aggregate the results.
```
Comment (jyemin) [inline driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala:179]: When do you use brackets and when to do you use backticks?

File: driver-scala/src/main/scala/org/mongodb/scala/model/Accumulators.scala [hunk 4]
```diff
@@ -167,14 +167,14 @@ object Accumulators {
...
-   * Creates an \$accumulator pipeline stage
+   * Creates an `\$accumulator` pipeline stage
    *
    * @param fieldName            the field name
    * @param initFunction         a function used to initialize the state
    * @param accumulateFunction   a function used to accumulate documents
    * @param mergeFunction        a function used to merge two internal states, e.g. accumulated on different shards or
    *                             threads. It returns the resulting state of the accumulator.
-   * @return the \$accumulator pipeline stage
+   * @return the `\$accumulator` pipeline stage
```
Comment (jyemin) [inline driver-scala/src/main/scala/org/mongodb/scala/model/Accumulators.scala:177]: Does the `$` still need to be escaped when it's inside of backticks?

File: driver-scala/src/main/scala/org/mongodb/scala/MongoCollection.scala (hunk 1 above)
Comment (rozza) [reply driver-scala/src/main/scala/org/mongodb/scala/MongoCollection.scala:154]: Not 100% sure but it solves the ambiguity warning that you saw previously.

File: driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala (hunk 2 above)
Comment (rozza) [reply driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala:177]: Yes good catch.

File: driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala (hunk 3 above)
Comment (rozza) [reply driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala:179]: Brackets are for links, backticks are for monospace text.

**Review by jyemin — APPROVED** [review]
lgtm

**Pull request merged**